### PR TITLE
policy: Add support for drop direction and reason

### LINKED ISSUE #1046: Add traffic drop sub reasons and do validation for those
Right now, when doing flow validation, the result only checks if the traffic is dropped: https://github.com/cilium/cilium-cli/blob/173587006fb0760e259ca4296db62dd6ed6f1406/connectivity/check/policy.go#L197-L198. It is not checking why and where exactly the traffic was dropped.

However, traffic can be dropped for multiple reasons such as routing problems, iptables, etc. Thus, we should have sub reasons for why a traffic is dropped so that we can be sure that the end result of a test is actually the one that we should be expecting to see.

Some examples are:
- Traffic dropped by a deny policy
- Traffic dropped at ingress of a pod
- Traffic dropped at egress of a pod

### PR DESCRIPTION
**Pull request by sayboras**
Please refer to individual commit for more details.

Fixes #1046

### CODE REVIEW

File: connectivity/suite.go [hunk 1]
```diff
@@ -345,13 +350,13 @@ func Run(ctx context.Context, ct *check.ConnectivityTest) error {
 		WithPolicy(echoIngressICMPDenyPolicyYAML). // Deny ICMP traffic from client to another client
 		WithFeatureRequirements(check.RequireFeatureEnabled(check.FeatureICMPPolicy)).
 		WithScenarios(
-			tests.PodToPod(),       // Client to echo traffic should be allowed
+			//tests.PodToPod(),       // Client to echo traffic should be allowed
```
Comment (tklauser) [inline connectivity/suite.go:353]: Why is this commented out?

File: connectivity/check/policy.go [hunk 2]
```diff
@@ -138,6 +139,20 @@ func deleteCNP(ctx context.Context, client *k8s.Client, cnp *ciliumv2.CiliumNetw
...
+	}
+
+	policyDenyReason = func(flow *flowpb.Flow) bool {
+		return flow.GetDropReasonDesc() == flowpb.DropReason_POLICY_DENY
+	}
+
+	defaultDenyReason = func(flow *flowpb.Flow) bool {
+		return flow.GetDropReasonDesc() == flowpb.DropReason_POLICY_DENIED
+	}
+)
```
Comment (tklauser) [inline connectivity/check/policy.go:154]: Any reason to not just make these plain `func`s?

File: connectivity/suite.go (hunk 1 above)
Comment (sayboras) [reply connectivity/suite.go:353]: sorry, it was unintentional as part of debugging :100:

File: connectivity/check/policy.go (hunk 2 above)
Comment (sayboras) [reply connectivity/check/policy.go]: Good catch :100:

**Review by aanm — APPROVED** [review]

**Comment by sayboras** [comment]
Force push to rebase with master and update drop result for newly added test (e.g. cluster-entity, host-identity, etc)

**Review by tklauser — APPROVED** [review]

**Pull request merged**